src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -448,11 +448,12 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
                 return typed(untpd.Apply(untpd.TypedSplice(arg), tree.expr), pt)
               case _ =>
             }
-          case tref: TypeRef if tref.symbol.isClass && !ctx.isAfterTyper =>
-            val setBefore = ctx.mode is Mode.GADTflexible
-            tpt1.tpe.<:<(pt)(ctx.addMode(Mode.GADTflexible))
-            if (!setBefore) ctx.retractMode(Mode.GADTflexible)
           case _ =>
+            if (!ctx.isAfterTyper) {
+              val setBefore = ctx.mode is Mode.GADTflexible
+              tpt1.tpe.<:<(pt)(ctx.addMode(Mode.GADTflexible))
+              if (!setBefore) ctx.retractMode(Mode.GADTflexible)
```

Comment:
This is from the GADT fix, where I didn't want to change the state of the context apart from relaxing the  solver for GADTs